src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.CompilationData.cs

```diff
@@ -64,60 +62,44 @@ public bool RemoveCachedSemanticModel(SyntaxTree tree)
 
             internal DeclarationAnalysisData GetOrComputeDeclarationAnalysisData(
                 SyntaxReference declaration,
-                Func<Func<DeclarationAnalysisDataBuilder>, DeclarationAnalysisDataBuilder> computeDeclarationAnalysisData,
+                Func<DeclarationAnalysisData> computeDeclarationAnalysisData,
                 bool cacheAnalysisData)
             {
-                DeclarationAnalysisDataBuilder dataBuilder = null;
-                try
+                if (!cacheAnalysisData)
                 {
-                    if (!cacheAnalysisData)
-                    {
-                        dataBuilder = computeDeclarationAnalysisData(_declarationAnalysisDataBuilderPool.Allocate);
-                        return DeclarationAnalysisData.CreateFrom(dataBuilder);
-                    }
+                    return computeDeclarationAnalysisData();
+                }
 
-                    DeclarationAnalysisData data;
-                    lock (_declarationAnalysisDataMap)
+                DeclarationAnalysisData data;
```

Comment:
why have this local at all.  just have something like "out var cachedData... return cachedData".   then, lower, do "var data = ...".   It seems super weird to want to share this same slot.